cmd/snap/cmd_run.go

```diff
@@ -1685,6 +1685,14 @@ func (x *cmdRun) runSnapConfine(info *snap.Info, runner runnable, beforeExec fun
 		env["KRB5CCNAME"] = krb5ccnamePath
 	}
 
+	// Guarantee that XDG_RUNTIME_DIR does exist before launching the snap
+	if xdg_runtime_dir, ok := env["XDG_RUNTIME_DIR"]; ok {
+		if err = os.Mkdir(xdg_runtime_dir, 0700); err != nil && !errors.Is(err, os.ErrExist) {
```

Comment:
should this have a unit test?

Reply:
Do you mean, a test to check what happens if it can't create the folder? Because there is already a test to ensure that the folder is created...